exp/controllers/rosamachinepool_controller_test.go

```diff
@@ -60,3 +79,245 @@ func TestNodePoolToRosaMachinePoolSpec(t *testing.T) {
+		err              error
+	)
+
+	setup := func(t *testing.T) {
+		t.Helper()
+		mockCtrl = gomock.NewController(t)
+		recorder = record.NewFakeRecorder(10)
+		ctx = context.TODO()
+		scheme = runtime.NewScheme()
+		ns, err = testEnv.CreateNamespace(ctx, "test-namespace")
+		g.Expect(err).To(BeNil())
+
+		g.Expect(expinfrav1.AddToScheme(scheme)).To(Succeed())
+		g.Expect(infrav1.AddToScheme(scheme)).To(Succeed())
+		g.Expect(clusterv1.AddToScheme(scheme)).To(Succeed())
+		g.Expect(expclusterv1.AddToScheme(scheme)).To(Succeed())
+		g.Expect(rosacontrolplanev1.AddToScheme(scheme)).To(Succeed())
+		g.Expect(corev1.AddToScheme(scheme)).To(Succeed())
+
+		secret = &corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "rosa-secret",
+				Namespace: ns.Name,
+			},
+			Data: map[string][]byte{
+				"ocmToken": []byte("secret-ocm-token-string"),
+			},
+		}
+
+		rosaControlPlane = &rosacontrolplanev1.ROSAControlPlane{
+			ObjectMeta: metav1.ObjectMeta{Name: "rosa-control-plane", Namespace: ns.Name},
+			TypeMeta: metav1.TypeMeta{
+				Kind:       "ROSAControlPlane",
+				APIVersion: rosacontrolplanev1.GroupVersion.String(),
+			},
+			Spec: rosacontrolplanev1.RosaControlPlaneSpec{
+				RosaClusterName:   "rosa-control-plane",
+				Subnets:           []string{"subnet-0ac99a6230b408813", "subnet-1ac99a6230b408811"},
+				AvailabilityZones: []string{"az-1", "az-2"},
+				Network: &rosacontrolplanev1.NetworkSpec{
+					MachineCIDR: "10.0.0.0/16",
+					PodCIDR:     "10.128.0.0/14",
+					ServiceCIDR: "172.30.0.0/16",
+				},
+				Region:           "us-east-1",
+				Version:          "4.15.20",
+				RolesRef:         rosacontrolplanev1.AWSRolesRef{},
+				OIDCID:           "iodcid1",
+				InstallerRoleARN: "arn1",
+				WorkerRoleARN:    "arn2",
+				SupportRoleARN:   "arn3",
+				CredentialsSecretRef: &corev1.LocalObjectReference{
+					Name: secret.Name,
+				},
+			},
+			Status: rosacontrolplanev1.RosaControlPlaneStatus{
+				Ready: true,
+				ID:    "rosa-control-plane1",
+			},
+		}
+
+		ownerCluster = &clusterv1.Cluster{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "capi-test-6",
+				Namespace: ns.Name,
+			},
+			Spec: clusterv1.ClusterSpec{
+				ControlPlaneRef: &corev1.ObjectReference{
+					Name:       rosaControlPlane.Name,
+					Kind:       "ROSAControlPlane",
+					APIVersion: rosacontrolplanev1.GroupVersion.String(),
+				},
+			},
+		}
+
+		ownerMachinePool = &expclusterv1.MachinePool{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "machinepool-1",
+				Namespace: ns.Name,
+				Labels:    map[string]string{clusterv1.ClusterNameLabel: ownerCluster.Name},
+				UID:       "owner-mp-uid-1",
+			},
+			TypeMeta: metav1.TypeMeta{
+				Kind:       "MachinePool",
+				APIVersion: expclusterv1.GroupVersion.String(),
+			},
+			Spec: expclusterv1.MachinePoolSpec{
+				ClusterName: ownerCluster.Name,
+				Template: clusterv1.MachineTemplateSpec{
+					Spec: clusterv1.MachineSpec{
+						ClusterName: ownerCluster.Name,
+					},
```

Comment:
We should add the infrastructureRef pointing to RosaMachinePool to have proper test,